tools/src/main/java/org/apache/kafka/tools/ConsoleShareConsumerOptions.java

```diff
@@ -0,0 +1,486 @@
+        }
+        return consumerProps;
+    }
+
+    private MessageFormatter buildFormatter() {
+        MessageFormatter formatter = null;
+        try {
+            Class<?> messageFormatterClass = Class.forName(options.valueOf(messageFormatterOpt));
+            formatter = (MessageFormatter) messageFormatterClass.getDeclaredConstructor().newInstance();
+
+            Properties formatterArgs = formatterArgs();
+            Map<String, String> formatterConfigs = new HashMap<>();
+            for (final String name : formatterArgs.stringPropertyNames()) {
+                formatterConfigs.put(name, formatterArgs.getProperty(name));
+            }
+
+            formatter.configure(formatterConfigs);
+
+        } catch (Exception e) {
+            CommandLineUtils.printUsageAndExit(parser, e.getMessage());
+        }
+        return formatter;
+    }
+
+    Properties consumerProps() {
+        return consumerProps;
+    }
+
+    boolean rejectMessageOnError() {
+        return options.has(rejectMessageOnErrorOpt);
+    }
+
+    AcknowledgeType acknowledgeType() {
+        // TODO : Once --release and --reject are implemented, change according to option
+        return AcknowledgeType.ACCEPT;
+    }
+
+    String topicArg() {
+        return options.valueOf(topicOpt);
+    }
+
+    int maxMessages() {
+        return options.has(maxMessagesOpt) ? options.valueOf(maxMessagesOpt) : -1;
+    }
+
+    int timeoutMs() {
+        return options.has(timeoutMsOpt) ? options.valueOf(timeoutMsOpt) : -1;
+    }
+
+    String bootstrapServer() {
+        return options.valueOf(bootstrapServerOpt);
+    }
+
+    Properties formatterArgs() throws IOException {
+        Properties formatterArgs = options.has(messageFormatterConfigOpt)
+                ? Utils.loadProps(options.valueOf(messageFormatterConfigOpt))
+                : new Properties();
+        String keyDeserializer = options.valueOf(keyDeserializerOpt);
+        if (keyDeserializer != null && !keyDeserializer.isEmpty()) {
+            formatterArgs.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer);
+        }
+        String valueDeserializer = options.valueOf(valueDeserializerOpt);
+        if (valueDeserializer != null && !valueDeserializer.isEmpty()) {
+            formatterArgs.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer);
+        }
+        formatterArgs.putAll(CommandLineUtils.parseKeyValueArgs(options.valuesOf(messageFormatterArgOpt)));
+
+        return formatterArgs;
+    }
+
+    MessageFormatter formatter() {
+        return formatter;
+    }
+
+    /**
+     * TEMPORARY - The classes below are defined as separate classes
+     * in the open PR - <a href="https://github.com/apache/kafka/pull/15274/files#">...</a>
+     * These inner classes are temporary and will be removed once the above PR gets merged.
+     */
+    public static class DefaultMessageFormatter implements MessageFormatter {
```

Comment:
This is guaranteeing a future build break. I suggest that you make a `"DefaultMessageFormatterTemporary"` as an inner class and use that temporarily instead.